news/php/news_article_gallery_feed.php

```diff
@@ -36,27 +36,54 @@ function create_news_article_gallery_feed($categories, $galleryStyle, $myBethel,
 //    $arrayOfNewsAndStories = get_xml($_SERVER["DOCUMENT_ROOT"] . "/_shared-content/xml/news-and-stories.xml", $categories, "inspect_news_article");
     $arrayOfNewsAndStories = sort_by_date($arrayOfNewsAndStories);
 
+    $includeStories = false;
+    $includeNews = false;
+
+    if( $newsOrStories == "Both" ) {
+        $includeStories = true;
+        $includeNews = true;
+    } elseif ( $newsOrStories == "News" ) {
+        $includeNews = true;
+    } else {
+        $includeStories = true;
+    }
+
     $threeStories = array();
-    foreach( $arrayOfNewsAndStories as $index => $newsAndStory){
+    foreach ($arrayOfNewsAndStories as $index => $newsAndStory) {
         $id = $newsAndStory['id'];
+        if( $includeNews ) {
+            // if its the Homepage Top Feature, skip any that aren't tagged as homepage
+            if( $galleryStyle == 'Homepage Top Feature' && !$newsAndStory['homepage-article'] )
+                continue;
+
+            if( $newsAndStory['article-type'] == 'News' and !in_array($id, $GLOBALS['stories-already-used'])) {
+                $newsAndStory['gallery-image'] = srcset($newsAndStory['image-path'], false, true, '', $newsAndStory['title']);
 
-        // if its the Homepage Top Feature, skip any that aren't tagged as homepage
-        if( $galleryStyle == 'Homepage Top Feature' && !$newsAndStory['homepage-article'] )
-            continue;
+                // don't use this story on this page again
+                array_push($GLOBALS['stories-already-used'], $id);
 
-        if( $newsAndStory['article-type'] == 'Story' and !in_array($id, $GLOBALS['stories-already-used'])){
-            $newsAndStory['gallery-image'] = srcset($newsAndStory['image-path'], false, true, '', $newsAndStory['title']);
+                array_push($threeStories, $newsAndStory);
+                unset($arrayOfNewsAndStories[$index]);
+            }
+        }
+        if( $includeStories ) {
+            // if its the Homepage Top Feature, skip any that aren't tagged as homepage
+            if( $galleryStyle == 'Homepage Top Feature' && !$newsAndStory['homepage-article'] )
+                continue;
 
-            // don't use this story on this page again
-            array_push($GLOBALS['stories-already-used'], $id);
+            if( $newsAndStory['article-type'] == 'Story' and !in_array($id, $GLOBALS['stories-already-used'])){
+                $newsAndStory['gallery-image'] = srcset($newsAndStory['image-path'], false, true, '', $newsAndStory['title']);
 
-            array_push($threeStories, $newsAndStory);
-            unset($arrayOfNewsAndStories[$index]);
+                // don't use this story on this page again
+                array_push($GLOBALS['stories-already-used'], $id);
 
-            // exit once there are 4
-            if( sizeof($threeStories) >= 4)
-                break;
+                array_push($threeStories, $newsAndStory);
+                unset($arrayOfNewsAndStories[$index]);
```

Comment:
if you wanted to, lines 63-66 and 77-81 are also doubled up - so you could consolidate

Reply:
I think it would make it really hard to read it if we made a method for just 3 lines of code.. would it?

Reply:
not necessarily a method. Maybe in each news/stories section you set a "add_this_thing = True", then down below do:
```
if( $add_this_thing == True )
    ...
```